### PR TITLE
feat(onboarding): Adding import seed screens for old&new users

### LINKED ISSUE #4955: Implement import seed phrase sequence
Depends on 
- https://github.com/status-im/status-desktop/issues/4954
- https://github.com/status-im/StatusQ/issues/567
- https://github.com/status-im/status-desktop/issues/4956
- [BIP39](https://github.com/bitcoin/bips/tree/master/bip-0039) is available

Specs:
- the autocomplete only searches the BIP39 wordlist
- the moment a user types the 3rd letter of a word, the word should be automatically completed and the focus should jump to the next word entry field.

### PR DESCRIPTION
**Pull request by alexandraB99**
Closes https://github.com/status-im/status-desktop/issues/4955
Closes https://github.com/status-im/status-desktop/issues/5230

### What does the PR do
Adds import seed phrase screens for onboarding a new user as well as for existing users

### Affected areas
Onboarding

### Screenshot of functionality
https://user-images.githubusercontent.com/31625338/159086349-9d5d918b-e741-425d-bed2-5be3be43b46a.mov

### CODE REVIEW

File: ui/app/AppLayouts/Onboarding/stores/BIP39_en.qml
Comment (saledjenic) [inline ui/app/AppLayouts/Onboarding/stores/BIP39_en.qml:3]: @alexandraB99 I am always for keeping such things on the backend rather (either having them hardcoded as it's done [here](https://github.com/status-im/status-desktop/blob/master/src/app_service/common/signing_phrases.nim) or load from the file on the backend) and expose that as list/model to qml. We should avoid mixing domain/logic and UI things.

**Review by saledjenic — APPROVED** [review]
@alexandraB99 changes done here are ok, approved. 

A comment below is something what would be good to update.

Apart of that I noticed some weird behaviour using keyboard for navigation in import seed screen:
- `tab` doesn't jump to another field, but updates text of the current field adding tab
- if there is suggestion list for a word I am adding `arrow down` jumps to another field instead of goes through the items of the list of suggestions

If user add custom name using capital letter once he is logged in this name is displayed completely in lower letters case.

@abhi-status @anastasiyaig not sure is that already logged as new issue or should be fixed through this one, that's why I didn't request changes.

File: ui/app/AppLayouts/Onboarding/stores/BIP39_en.qml
Comment (alexandraB99) [reply ui/app/AppLayouts/Onboarding/stores/BIP39_en.qml:3]: @saledjenic yes, we have discussed this with Rabbits and concluded to put the model in QML

**Comment by alexandraB99** [comment]
> @alexandraB99 changes done here are ok, approved.
> 
> A comment below is something what would be good to update.
> 
> Apart of that I noticed some weird behaviour using keyboard for navigation in import seed screen:
> 
> * `tab` doesn't jump to another field, but updates text of the current field adding tab
> * if there is suggestion list for a word I am adding `arrow down` jumps to another field instead of goes through the items of the list of suggestions
> 
> If user add custom name using capital letter once he is logged in this name is displayed completely in lower letters case.
> 
> @abhi-status @anastasiyaig not sure is that already logged as new issue or should be fixed through this one, that's why I didn't request changes.

@saledjenic Tab and down arrow keys behavior is not specified, hence not taken into account. @John-44 what should tab and down arrow keys do when pressed in the input seed phrase view?

also regarding the other comment, "If user add custom name using capital letter once he is logged in this name is displayed completely in lower letters case." This PR is not touching the display name part, so I am not sure what is supposed to be stored and how. Maybe @richard-ramos could give some input?

**Comment by abhi-status** [comment]
> If user add custom name using capital letter once he is logged in this name is displayed completely in lower letters case.

@saledjenic Thanks for letting me know. 
That profile name screen is just a temporary one until #4985 (still WIP) gets merged. I'll keep a note of this issue and test once 4985 is ready to test raise it there if it still exists.

**Comment by abhi-status** [comment]
@John-44 @alexandraB99 Do we need an import seed phrase flow for new users ?
Isn't this only meant for existing users who already have a seed phrase saved somewhere ?
I ask, because I was under the impression that this new component was only meant for existing users.

File: ui/app/AppLayouts/Onboarding/views/SeedPhraseInputView.qml
Comment (noeliaSD) [inline ui/app/AppLayouts/Onboarding/views/SeedPhraseInputView.qml:16]: Suggestion: `OnboardingBasePage` has the `finished()` signal that maybe could be used instead of `exit()` one. Indeed it is for the same purpose!

File: ui/app/AppLayouts/Onboarding/views/SeedPhraseInputView.qml
Comment (noeliaSD) [inline ui/app/AppLayouts/Onboarding/views/SeedPhraseInputView.qml:57]: Suggestion: Could it make sense to have `States` depending on the selected tab bar and change the needed properties in transitions (like the model and maybe the construction of the SeedPhraseInput grid too)? So maybe it'd be more clear than selecting the model by subtracting a piece of the string text.

**Review by noeliaSD — CHANGES_REQUESTED** [review]
@alexandraB99 the new screen looks super! :)

I just have some questions and suggestions:

1. I've tested it and when changing the tab does not happen nothing. I don't know if it is out of the scope or it is something broken but I was expecting the sheed frase inputs grid to change from 12 to 18 or 24 inputs.
2. Could it be possible to create some `readonly properties` for values that have a sense and maybe are repeated during the code? Like for example `12` or `18` when it refers to the amount of words.
3. And the last thing.. I attach a video for better understanding but basically what I can see is that sometimes the word is autocompleted and sometimes not (I cannot see the pattern to give more info..). Also, sometimes it jumps to the next input and other times the user needs to click on the next one. Please, could you take a look on that?

https://user-images.githubusercontent.com/97019400/159690897-6ff17b71-0353-4a8d-8dae-476ac9173426.mov

**Comment by abhi-status** [comment]
There is this weird issue on Windows, where each word field in the seed phrase seems to add a new line and a small scroll bar on the right.

https://user-images.githubusercontent.com/95302135/159775218-8841d4aa-281b-4311-8711-9048dc30534c.mp4

**Comment by abhi-status** [comment]
Also, the dropdown field gets blocked by the Import button
image

**Comment by alexandraB99** [comment]
> > . @John-44 what should tab and down arrow keys do when pressed in the input seed phrase view?
> 
> @alexandraB99 pressing the tab key should jump the focus to the next input field.
> 
> Pressing the down (or up) arrow key could move the focus to the field directly above or below (or do nothing, if there is no field above/below)

@John-44 adde key navigation, for tab, up/down, as well as right/left. The tab behavior is currently a bit problematic but I am looking into this

https://user-images.githubusercontent.com/31625338/160014263-7c57f769-dd7e-4f96-93c9-c1de8729e0dc.mov

File: ui/app/AppLayouts/Onboarding/views/SeedPhraseInputView.qml
Comment (alexandraB99) [reply ui/app/AppLayouts/Onboarding/views/SeedPhraseInputView.qml:57]: where could States be used here, I don't quite get it? this is setting the model, thus no State changes, but just the model number, so how many delegates of the same component we will show

Comment (noeliaSD) [reply ui/app/AppLayouts/Onboarding/views/SeedPhraseInputView.qml:57]: > 3\. And the last thing.. I attach a video for better understanding but basically what I can see is that sometimes the word is autocompleted and sometimes not (I cannot see the pattern to give more info..). Also, sometimes it jumps to the next input and other times the user needs to click on the next one. Please, could you take a look on that?

@John-44, @alexandraB99 explained me the requirements are the following: 
```
- the autocomplete only searches the BIP39 wordlist
- the moment a user types the 3rd letter of a word, the word should be automatically completed and the focus should jump to the next word entry field.
```
But when testing it, as a user I was a bit confused because I expected the same behavior on each word input, regardless of whether it matches 3 letters or needs more. So I expected that it should be also autocompleted when there are NO MORE options regardless the number of chars that are introduced. I.e: `hammer`, when introducing `hamm` I think it should be as well autocompleted. 

Maybe there is a specific reason to only autocomplete if there is 3 characters/unique match, but like I said, as a UX perspective, for me it was confusing!

**Comment by richard-ramos** [comment]
image

Emoji hash and identicon ring is not being shown. Also the chatId is incorrect. Should be 130 characters long

**Comment by richard-ramos** [comment]
image
When clicking on "Add another existing key", the old seed phrase dialog is shown

**Comment by abhi-status** [comment]
> image
> 
> When clicking on "Add another existing key", the old seed phrase dialog is shown

@richard-ramos This PR deals with the flow for only _new_ users

**Comment by alexandraB99** [comment]
> image
> 
> Emoji hash and identicon ring is not being shown.

@richard-ramos please check again, this issue should be fixed now

**Comment by alexandraB99** [comment]
> > > . @John-44 what should tab and down arrow keys do when pressed in the input seed phrase view?
> > 
> > 
> > @alexandraB99 pressing the tab key should jump the focus to the next input field.
> > Pressing the down (or up) arrow key could move the focus to the field directly above or below (or do nothing, if there is no field above/below)
> 
> @John-44 added key navigation, for tab, up/down, as well as right/left. The tab behavior is currently a bit problematic but I am looking into this
> 
>  keynav.mov

tab key also fixed

**Comment by alexandraB99** [comment]
> Also, the dropdown field gets blocked by the Import button
> image

@abhi-status  can't reproduce this, I see the list on top of the button (red color is for emphasis)
sugList

**Comment by abhi-status** [comment]
On Windows, there is also the issue of not being able to import a profile using a valid seed phrase.

image

**Comment by abhi-status** [comment]
I can now see a similar issue with the StatusQ component on the Contacts list in Create Group Chat window as well 
--> https://github.com/status-im/status-desktop/issues/5221

**Review by noeliaSD — APPROVED** [review]

**Comment by noeliaSD** [comment]
LGTM! :)

**Comment by abhi-status** [comment]
Hey @alexandraB99 I am not sure if the fix - #5255 is here, as I can still seem to import other accounts 
(via the import seed phrase flow)

Could you please rebase ?

**Comment by abhi-status** [comment]
Using the arrow keys on the keyboard : 

- User cannot select the suggested seed words (in the dropdown) using down key

- Using left/right keys does not traverse through all the letters in the word

- the suggestion dropdown appears randomly behind the word field (at ~35sec in the video 👇)

https://user-images.githubusercontent.com/95302135/161075656-12094ec8-2307-4fb5-ba16-9f2f1158783a.mp4

**Review by iurimatias — APPROVED** [review]

**Comment by iurimatias** [comment]
can confirm, issue of mnemonic has been fixed, the other issue raised is not the desired behaviour

**Pull request merged**
